query-engine/schema/src/identifier_type.rs

```diff
@@ -186,7 +186,12 @@ impl std::fmt::Display for IdentifierType {
             IdentifierType::ToOneRelationFilterInput(related_model, arity) => {
                 let nullable = if arity.is_optional() { "Nullable" } else { "" };
 
-                write!(f, "{}{}RelationFilter", capitalize(related_model.name()), nullable)
+                write!(
+                    f,
+                    "{}{}ScalarRelationFilter",
```

Comment:
The only tests I expect to break in `prisma/prisma` are type snapshots in `@prisma/client`.